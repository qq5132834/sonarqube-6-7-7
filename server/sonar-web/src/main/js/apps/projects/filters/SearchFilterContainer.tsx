/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import * as PropTypes from 'prop-types';
import { debounce } from 'lodash';
import { getFilterUrl } from './utils';
import SearchFilter from './SearchFilter';

interface Props {
  className?: string;
  query: { search?: string };
  isFavorite?: boolean;
  organization?: { key: string };
}

export default class SearchFilterContainer extends React.PureComponent<Props> {
  static contextTypes = {
    router: PropTypes.object.isRequired
  };

  constructor(props: Props) {
    super(props);
    this.handleSearch = debounce(this.handleSearch, 250);
  }

  handleSearch = (userQuery?: string) => {
    const path = getFilterUrl(this.props, { search: userQuery });
    this.context.router.push(path);
  };

  render() {
    return (
      <SearchFilter
        className={this.props.className}
        query={this.props.query}
        handleSearch={this.handleSearch}
      />
    );
  }
}
