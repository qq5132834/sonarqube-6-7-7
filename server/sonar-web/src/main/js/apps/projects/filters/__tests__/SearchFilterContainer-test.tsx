/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import SearchFilterContainer from '../SearchFilterContainer';

// mocking lodash, because mocking timers is now working for some reason :'(
jest.mock('lodash', () => {
  const lodash = require.requireActual('lodash');
  lodash.debounce = (fn: Function) => (...args: any[]) => fn(args);
  return lodash;
});

it('searches', () => {
  const push = jest.fn();
  const wrapper = shallow(<SearchFilterContainer query={{}} />, { context: { router: { push } } });
  expect(wrapper).toMatchSnapshot();
  wrapper.prop('handleSearch')('foo');
  expect(push).toBeCalledWith({ pathname: '/projects', query: { search: 'foo' } });
});
