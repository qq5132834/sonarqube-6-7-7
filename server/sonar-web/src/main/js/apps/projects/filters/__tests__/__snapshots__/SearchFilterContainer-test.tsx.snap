// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searches 1`] = `
<SearchFilter
  handleSearch={[Function]}
  query={Object {}}
/>
`;
